cmd/clusterctl/client/cluster/workload_cluster.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"fmt"
+
+	kc "sigs.k8s.io/cluster-api/util/kubeconfig"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// WorkloadCluster has methods for fetching kubeconfig of workload cluster from management cluster.
+type WorkloadCluster interface {
+	//Get workload cluster kubeconfig
+	GetKubeconfig(name string) error
+}
+
+// workloadCluster implements WorkloadCluster.
+type workloadCluster struct {
+	proxy Proxy
+}
+
+func (p *workloadCluster) GetKubeconfig(name string) error {
+	cs, err := p.proxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	obj := client.ObjectKey{
+		Namespace: "default",
```

Comment:
We can't hard-code the namespace because clusters can exist in any namespace. You'll need to pass in the namespace along with the name.

Reply:
For sake of consistency with other methods:
- there should be a --namespace flag
- if the flag value is empty, we should try to use the current namespace as default
- if this is not possible, we should ask explicitly to the user to specify a namespace

This is how it is implemented in another clusterctl method https://github.com/kubernetes-sigs/cluster-api/blob/e87da61eaec4eeb5d0cd702099e2b8380b2ba719/cmd/clusterctl/client/config.go#L174-L184